scripts/mailing_lists/migrate_project_mailing_lists.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+"""Create mailing lists for all top-level projects
+"""
+import logging
+import sys
+
+from modularodm import Q
+
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from website.mailing_list.utils import create_list as real_create_list
+from website.models import Node
+from website.notifications.model import NotificationSubscription
+from website.notifications.utils import to_subscription_key
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+EVENT = 'mailing_list_events'
+SUBSCRIPTION_TYPE = 'email_transactional'
+
+def fake(*args, **kwargs):
+    # Override actual method to prevent outgoing calls
+    return
+
+def get_targets():
+    return Node.find()
+
+def migrate(dry_run=True):
+    if dry_run:
+        create_list = fake
+    else:
+        create_list = real_create_list
+
+    successful_enables = []
+    successful_disables = []
+    unknown_failures = {}
+    nodes = get_targets()
+    ncount = len(nodes)
+    logger.info('Preparing to migrate {} nodes.'.format(ncount))
+    for i, node in enumerate(nodes):
+        if not node.is_mutable_project:
+            try:
+                logger.info('({0}/{1})Disabling mailing list for registration/dashboard {2}'.format(i+1, ncount, node._id))
+                node.mailing_enabled = False    
+                node.mailing_updated = False
+                node.save()
+                successful_disables.append(node._id)
+            except Exception as e:
+                logger.exception('Error while handling node {}'.format(node._id))
+                unknown_failures[node._id] = e
+        else:
+            try:
+                subscription = NotificationSubscription(
+                    _id=to_subscription_key(node._id, EVENT),
+                    owner=node,
+                    event_name=EVENT
+                )
+                logger.info('({0}/{1})Enabling mailing list for node {2}'.format(i+1, ncount, node._id))
+                node.mailing_enabled = True
+                node.mailing_updated = True
+
+                for user in node.contributors:
+                    if user.is_active:
+                        logger.info('Subscribing user {} on node {}'.format(user, node))
+                        subscription.add_user_to_subscription(user, SUBSCRIPTION_TYPE)
+                        # users added on `create_list`
+
+                subscription.save()
+                node.save()
+                successful_enables.append(node._id)
+            except Exception as e:
+                logger.exception('Error while handling node {}'.format(node._id))
+                unknown_failures[node._id] = e
+            else:
+                create_list(node._id)
```

Comment:
As far as I can tell from their documentation, we shouldn't hit a rate limit. Their Webhooks and Domains APIs are rate limited, but the docs don't say that the Mailing Lists API is. As for state, that isn't explicitly handled. The simplest solution would be to just not roll out the UI changes until after the migration.